src/styles/utils/functions.scss

```diff
@@ -0,0 +1,3 @@
+@function dynamicMaxWidth($width, $margins, $amount) {
+  @return $width * $amount + $margins * ($amount - 1);
+}
```

Comment:
I would name this function `getMaxWidth` or something. It will be maximally clear this way